runtime/src/test/java/com/vaadin/quarkus/context/AbstractContextTest.java

```diff
@@ -0,0 +1,258 @@
+
+        T referenceA = context.get(contextual, creationalContext);
+        referenceA.setState("hello");
+
+        context.destroy(contextual);
+        assertEquals(1, destroyedBeans.size());
+    }
+
+    @Test
+    public void destroyQuarkusContext_beanExistsInContext_beanDestroyed() {
+        createContext().activate();
+
+        T referenceA = context.get(contextual, creationalContext);
+        referenceA.setState("hello");
+
+        context.destroy();
+        assertEquals(1, destroyedBeans.size());
+    }
+
+    @Test
+    public void destroyAllActive_beanExistsInContext_beanDestroyed() {
+        createContext().activate();
+
+        T referenceA = context.get(contextual, creationalContext);
+        referenceA.setState("hello");
+
+        context.destroyAllActive();
+        assertEquals(1, destroyedBeans.size());
+    }
+
+    @Test
+    public void destroyAllActive_severalContexts_beanDestroyed() {
+        UnderTestContext contextUnderTestA = createContext();
+        contextUnderTestA.activate();
+        T referenceA = context.get(contextual, creationalContext);
+
+        ContextualStorage storageA = context.getContextualStorage(contextual,
+                false);
+
+        referenceA.setState("hello");
+
+        final UnderTestContext contextUnderTestB = createContext();
+        contextUnderTestB.activate();
+        T referenceB = context.get(contextual, creationalContext);
+
+        referenceB.setState("foo");
+
+        ContextualStorage storageB = context.getContextualStorage(contextual,
+                false);
+
+        assertEquals(2, createdBeans);
+
+        AbstractContext.destroyAllActive(storageA);
+        assertEquals(1, destroyedBeans.size());
+
+        AbstractContext.destroyAllActive(storageB);
+        assertEquals(2, destroyedBeans.size());
+    }
+
+    @Test
+    public void getState_beanExistsInContext_contextualInstanceAndBeanAreReturned() {
+        createContext().activate();
+
+        T reference = context.get(contextual, creationalContext);
+        reference.setState("hello");
+
+        ContextState state = context.getState();
+
+        Map<InjectableBean<?>, Object> instances = state
+                .getContextualInstances();
+        assertNotNull(instances);
+        assertSame(reference, instances.get(contextual));
+    }
+
+    protected UnderTestContext createContext() {
+        UnderTestContext underTestContext = newContextUnderTest();
+        /*
+         * UnderTestContext implementations set fields to Vaadin
+         * CurrentInstance. Need to hold a hard reference to prevent possible
+         * GC, because CurrentInstance works with weak reference.
+         */
+        contexts.add(underTestContext);
+        return underTestContext;
+    }
+
+    protected abstract UnderTestContext newContextUnderTest();
+
+    protected abstract boolean isNormalScoped();
+
+    protected abstract Class<T> getBeanType();
+
+    protected abstract Class<C> getContextType();
```

Comment:
All the abstracts should have a javadoc with a minimal on what is expected of the implementation

Reply:
this is a test class which is modified copy of https://github.com/vaadin/cdi/blob/master/vaadin-cdi/src/test/java/com/vaadin/cdi/context/AbstractContextTest.java.
The latter class has no any javadocs.